lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -391,6 +392,46 @@ def allocate_load_balancer(action_handler, lb_spec, lb_options, machine_specs)
         end
       end
 
+      # Update the load balancer sticky sessions
+      if sticky_sessions
+        policies = elb.client.describe_load_balancer_policies(load_balancer_name: actual_elb.name)
+        current = policies[:policy_descriptions].detect { |pd| pd[:policy_type_name] == 'AppCookieStickinessPolicyType' }
+
+        if current.nil?
+          current_cookie_hash = {}
+        else
+          current_cookie_hash = current[:policy_attribute_descriptions].detect { |pad| pad[:attribute_name] == 'CookieName' }
+        end
+
+        policy_name = "#{actual_elb.name}-sticky-session-policy"
+
+        if current_cookie_hash[:attribute_value] != sticky_sessions[:cookie_name]
+          perform_action.call("  updating sticky sessions to #{sticky_sessions[:cookie_name].inspect}") do
+            count = 0
+
+            begin
+              elb.client.create_app_cookie_stickiness_policy(
+                load_balancer_name: actual_elb.name,
+                policy_name: policy_name,
+                cookie_name: sticky_sessions[:cookie_name]
+              )
+            rescue AWS::ELB::Errors::DuplicatePolicyName
+              elb.client.delete_load_balancer_policy(
+                load_balancer_name: actual_elb.name,
+                policy_name: policy_name
+              )
+
+              if count < 1
+                count += 1
```

Comment:
Should we guard the delete with this count?  If we get duplicate_policy_name again, is deleting it before raising the right thing?